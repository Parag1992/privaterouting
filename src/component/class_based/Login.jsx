import React from "react";
import { connect } from "react-redux";
import { Navigate } from "react-router-dom";
import postCall from "./apiCall";
import { saveuserdetailsAct } from "../../Actions";

class Login extends React.Component {

    constructor(props)
    {
        super(props);
        this.state={
            isLogin:false
        }
    }
    UpdateUsrNmTxt = (e) =>{
        console.log(e.target.value);
    }

    UpdatePassTxt = (e) =>{
        console.log(e.target.value);
    }
    
    onBtnClicked = () =>{
        
        postCall({username: 'kminchelle',
        password: '0lelplR'})
        .then(response=>{
            response.data.token && this.setState({isLogin:true});
            this.props.saveusrdetails(response.data);
        })
        .catch(err=>console.log('post call err',err))
        
    }

    render(){

        return (
            <div>
                   <label>Login Form</label><br/>
                   <label>User Name</label>  <input type={"text"} className={"txtUsrCls"} key={"txtUsr"} onChange={this.UpdateUsrNmTxt} /><br/>
                   <label>Password</label>  <input type={"password"} className={"txtPassCls"} key={"txtPass"} onChange={this.UpdatePassTxt} /><br/>
                   <button className="myBtnCls" value={"Submit"} onClick={this.onBtnClicked} >Submit</button>
                   {this.state.isLogin && <Navigate to={'/home'}/>}
            </div>
        )
    }
}

const mapStateToProps = (props) =>{
  return {
      userdetails:props.saveuserdetails
  }
}

const mapDispatchToProps = (dispatch) =>{
    return {
        saveusrdetails:(data) =>dispatch(saveuserdetailsAct(data))
    }
}
export default connect(mapStateToProps,mapDispatchToProps) (Login);