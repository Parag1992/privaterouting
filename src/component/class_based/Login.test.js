import { shallow } from "enzyme"
import Login from "./Login"

describe('Test login component',()=>{


    test('Check for input box',()=>{
        let wrapper=shallow(<Login/>);

        expect(wrapper.exists('.txtPassCls')).toBe(true);
        expect(wrapper.exists('.txtUsrCls')).toBe(true);
    })

    test('Check Initial state',()=>{
        let wrapper=shallow(<Login/>);

        expect(wrapper.state()).toEqual({isLogin:false});
    })

    test('snapshot testing',()=>{
        let wrapper=shallow(<Login/>);

        expect(wrapper).toMatchSnapshot();
    })
})