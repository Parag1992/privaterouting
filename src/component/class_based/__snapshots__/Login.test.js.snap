// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test login component snapshot testing 1`] = `
<div>
  <label>
    Login Form
  </label>
  <br />
  <label>
    User Name
  </label>
    
  <input
    className="txtUsrCls"
    key="txtUsr"
    onChange={[Function]}
    type="text"
  />
  <br />
  <label>
    Password
  </label>
    
  <input
    className="txtPassCls"
    key="txtPass"
    onChange={[Function]}
    type="password"
  />
  <br />
  <button
    className="myBtnCls"
    onClick={[Function]}
    value="Submit"
  >
    Submit
  </button>
</div>
`;
